src/coreclr/System.Private.CoreLib/src/System/StubHelpers.cs

```diff
@@ -1241,7 +1241,8 @@ internal static IntPtr SafeHandleAddRef(SafeHandle pHandle, ref bool success)
         {
             if (pHandle == null)
             {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.pHandle, ExceptionResource.ArgumentNull_SafeHandle);
+                success = false;
```

Comment:
Did you consider instead having this be true? Then Release would just nop for null, and you wouldn't emit the branch in the emitted code.

Reply:
Hmm, maybe I've misunderstood the flow. I thought the emitted IL was referencing this, but it appears that code is calling DangerousAddRef directly
 When is this used?

Reply:
@stephentoub Yeah, one would think. For CrossGen2 your statement is accurate but these functions aren't used there - see https://github.com/dotnet/runtime/pull/47944#issuecomment-774428059 above. For the built-in IL Stubs these APIs are used and the branch is on the `success` by-ref not the `null` value - see below. It would be possible to update the generated IL Stub but I would prefer to handle the changes here since I would need to change more code to directly emit calls to the `SafeHandle.Dangerous*` functions.

https://github.com/dotnet/runtime/blob/a3cbb03a29ccea072e34d8bad8bb95d77881a210/src/coreclr/vm/ilmarshalers.cpp#L2784-L2809